src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.stories.js

```diff
@@ -0,0 +1,205 @@
+import {useArgs} from '@storybook/client-api';
+import {expect, jest} from '@storybook/jest';
+import {userEvent, within} from '@storybook/testing-library';
+
+import {FormDecorator} from 'components/admin/form_design/story-decorators';
+import Field from 'components/admin/forms/Field';
+import Fieldset from 'components/admin/forms/Fieldset';
+import FormRow from 'components/admin/forms/FormRow';
+
+import ObjectsApiOptionsFormFields from './ObjectsApiOptionsFormFields';
+import {mockObjecttypeVersionsGet, mockObjecttypesGet} from './mocks';
+
+const render = ({index, label, name}) => {
+  const [{formData}, updateArgs] = useArgs();
+  const onChange = newValues => {
+    updateArgs({formData: newValues});
+  };
+
+  return (
+    <Fieldset>
+      <FormRow>
+        <Field name={name} label={label}>
+          <ObjectsApiOptionsFormFields
+            index={index}
+            name={name}
+            formData={formData}
+            onChange={onChange}
+          />
+        </Field>
+      </FormRow>
+    </Fieldset>
+  );
+};
+
+export default {
+  title: 'Form design/Registration/Objects API',
+  decorators: [FormDecorator],
+  render,
+  args: {
+    formData: {},
+  },
+  parameters: {
+    msw: {
+      handlers: [
+        mockObjecttypesGet([
+          {
+            url: 'https://objecttypen.nl/api/v1/objecttypes/2c77babf-a967-4057-9969-0200320d23f1',
+            uuid: '2c77babf-a967-4057-9969-0200320d23f1',
+            name: 'Tree',
+            namePlural: 'Trees',
+            dataClassification: 'open',
+          },
+          {
+            url: 'https://objecttypen.nl/api/v1/objecttypes/2c77babf-a967-4057-9969-0200320d23f2',
+            uuid: '2c77babf-a967-4057-9969-0200320d23f2',
+            name: 'Person',
+            namePlural: 'Persons',
+            dataClassification: 'open',
+          },
+        ]),
+        mockObjecttypeVersionsGet([
+          {version: 1, status: 'published'},
+          {version: 2, status: 'draft'},
+        ]),
+      ],
+    },
+  },
+};
+
+export const Default = {};
+
+export const SwitchToV2Empty = {
+  play: async ({canvasElement}) => {
+    window.confirm = jest.fn(() => true);
+    const canvas = within(canvasElement);
+
+    const v2Tab = canvas.getByRole('tab', {selected: false});
+    userEvent.click(v2Tab);
+
+    await waitFor(async () => {
+      const objecttypeSelect = canvas.getByLabelText('Objecttype');
+
+      await expect(objecttypeSelect.value).toBe(
+        'https://objecttypen.nl/api/v1/objecttypes/2c77babf-a967-4057-9969-0200320d23f1'
+      );
+    });
+
+    await waitFor(async () => {
+      const objecttypeVersionSelect = canvas.getByLabelText('Objecttype version');
+
+      await expect(objecttypeVersionSelect.value).toBe('2');
+    });
+
+    const v1Tab = canvas.getByRole('tab', {selected: false});
+    userEvent.click(v1Tab);
```

Comment:
```suggestion
    await userEvent.click(v1Tab);
```
backend is on a recent storybook version where testing-library/user-events is everything async